### ISSUE TITLE
Icon preview in technology and focus trees not always showing with gfxIndex enabled or with dependencies set.

### ISSUE THREAD

**Issue by BiscuitCookies** [herbix/hoi4modutilities#88]
I have gfxIndex featureflag enabled.
So it seems to in some cases not read in load order or not at all

For example for La resistance the armored car
in the preview even with dependencies set:
![image](https://github.com/herbix/hoi4modutilities/assets/25040791/a23a4dc5-87c2-41eb-b727-ef9e478cf5d6)
It does load a modified icon of the same name if I place one in the mod folder
![image](https://github.com/herbix/hoi4modutilities/assets/25040791/da9eb37f-52db-46cf-8241-60458c147c2e)
when gfx define removed from the dlc file but still set in the mod file
![image](https://github.com/herbix/hoi4modutilities/assets/25040791/80373d44-e4bd-463f-9966-7148cf0a10ba)
in the game:
![image](https://github.com/herbix/hoi4modutilities/assets/25040791/bf3f1dea-d6b6-4e3e-8eb2-a5933c9c84f4)

Cases of gfxIndex not working
With gfxIndex enabled
![image](https://github.com/herbix/hoi4modutilities/assets/25040791/e4b47686-3f4b-4f34-838e-5ecc68b65e04)
when dependencies specifically set:
![image](https://github.com/herbix/hoi4modutilities/assets/25040791/9d633281-2f41-4eb0-bb0b-a1355a5141e6)

And these are only examples for the technology tree

**Comment by herbix** [comment]
Does reopening the preview fix this?
There's a known limitation that gfx index needs tome to load. And it won't update when .gfx in your mod folder changes.

**Comment by BiscuitCookies** [comment]
No, it does not. And these are were checked when the building of the gfx index is done notification has popped up.
There are sometimes cases where doing a edit in the file causes the preview to show icons that were previously not shown, but it doesn't always do that

**Comment by BiscuitCookies** [comment]
I did notice browsing through the code that there is no mention of gfindex in the [src/previewdef/technology/loader.ts](https://github.com/herbix/hoi4modutilities/blob/master/src/previewdef/technology/loader.ts) file.
Which still wouldn't explain why certain icons aren't being overridden by later defined icons or not shown at all.

**Comment by herbix** [comment]
gfxindex usage is here: https://github.com/herbix/hoi4modutilities/blob/6a9201de7a3d57b9ded17cf3a37c843294a39514/src/util/image/imagecache.ts#L39

**Comment by BiscuitCookies** [comment]
I meant I didn't see these in technology
https://github.com/herbix/hoi4modutilities/blob/6a9201de7a3d57b9ded17cf3a37c843294a39514/src/previewdef/focustree/loader.ts#L6
https://github.com/herbix/hoi4modutilities/blob/6a9201de7a3d57b9ded17cf3a37c843294a39514/src/previewdef/focustree/loader.ts#L34

but I might be completely wrong on what those do.

**Comment by BiscuitCookies** [comment]
> Does reopening the preview fix this? There's a known limitation that gfx index needs time to load. And it won't update when .gfx in your mod folder changes.

Odd behavior I found is that when you reload the entire vscode window and when you've set the gfx dependencies in the file.
It does show the correct images, but when you then make any changes to the technology file the incorrect icons or no icons will show again.
